tools/etcdhelper/etcdhelper.go

```diff
@@ -0,0 +1,128 @@
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	etcd "github.com/coreos/etcd/clientv3"
+	"golang.org/x/net/context"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime/serializer/json"
+	"k8s.io/kubernetes/pkg/runtime/serializer/protobuf"
+
+	// install all APIs
+	_ "github.com/openshift/origin/pkg/api/install"
+	_ "github.com/openshift/origin/pkg/quota/api/install"
+	_ "k8s.io/kubernetes/pkg/api/install"
+)
+
+func main() {
+	var endpoint, keyFilePath, certFilePath, caFilePath string
+	flag.StringVar(&endpoint, "endpoint", "https://127.0.0.1:4001", "Etcd endpoint.")
+	flag.StringVar(&keyFilePath, "key-file", "", "TLS client key.")
+	flag.StringVar(&certFilePath, "cert-file", "", "TLS client certificate.")
+	flag.StringVar(&caFilePath, "ca-file", "", "Server TLS CA certificate.")
+	flag.Parse()
+
+	if flag.NArg() == 0 {
+		fmt.Fprintf(os.Stderr, "ERROR: you need to specify action ls [<key>] or get <key>\n")
+		os.Exit(1)
+	}
+	if flag.Arg(0) == "get" && flag.NArg() == 1 {
+		fmt.Fprintf(os.Stderr, "ERROR: you need to specify <key> for get operation\n")
+		os.Exit(1)
+	}
+	action := flag.Arg(0)
+	key := ""
+	if flag.NArg() > 1 {
+		key = flag.Arg(1)
+	}
+
+	config := etcd.Config{
+		Endpoints: []string{endpoint},
+		TLS: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+	if len(certFilePath) != 0 && len(keyFilePath) != 0 {
+		cert, err := tls.LoadX509KeyPair(certFilePath, keyFilePath)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR: unable to read cert-file or key-file pair: %v\n", err)
+			os.Exit(1)
+		}
+		config.TLS.Certificates = []tls.Certificate{cert}
+	}
+	if len(caFilePath) != 0 {
+		caFile, err := ioutil.ReadFile(caFilePath)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "ERROR: unable to read ca-file: %v\n", err)
+			os.Exit(1)
+		}
+		rootCAs := x509.NewCertPool()
+		rootCAs.AppendCertsFromPEM(caFile)
+		config.TLS.InsecureSkipVerify = false
+		config.TLS.RootCAs = rootCAs
+	}
+
+	client, err := etcd.New(config)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: unable to connect to etcd: %v\n", err)
+		os.Exit(1)
+	}
+	defer client.Close()
+
+	switch action {
+	case "ls":
+		err = listKeys(client, key)
+	case "get":
+		err = getKey(client, key)
+	}
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: %s-ing %s: %v\n", action, key, err)
+		os.Exit(1)
+	}
+}
+
+func listKeys(client *etcd.Client, key string) error {
+	var resp *etcd.GetResponse
+	var err error
+	if len(key) == 0 {
+		resp, err = client.Get(context.TODO(), "/", etcd.WithFromKey(), etcd.WithKeysOnly())
+	} else {
+		resp, err = client.Get(context.TODO(), key, etcd.WithPrefix(), etcd.WithKeysOnly())
+	}
+	if err != nil {
+		return err
+	}
+	for _, ev := range resp.Kvs {
+		fmt.Printf("%s\n", ev.Key)
+	}
+	return nil
+}
+
+func getKey(client *etcd.Client, key string) error {
+	resp, err := client.Get(context.TODO(), key)
+	if err != nil {
+		return err
+	}
+	ps := protobuf.NewSerializer(api.Scheme, api.Scheme, "application/vnd.kubernetes.protobuf")
```

Comment:
It is meant as a dev helper to get raw contents straight out of etcd, without the need to changing openshift configs how objects are stored. Previously you could get raw json straight from etcd, now you can't. This is the replacement.